### PR TITLE
Add auto approval and labels support

### PR DESCRIPTION
**Pull request by janboll**
Related schema change: https://github.com/app-sre/qontract-schemas/pull/626

[APPSRE-10065](https://issues.redhat.com/browse/APPSRE-10065)

### CODE REVIEW

**Comment by janboll** [comment]
> Please elaborate on the relationship between the template-collection `enableAutoApproval` and the template `autoApproved`. Do I need to set all to `true` for an automatic merge?

`enableAutoApproval` on the collection turns on auto approval in general. Can turn it off, i.e. after dress-up is done.
`autoApproved` marks a template to be auto-approved.

Auto-approved changes are grouped together and added to one MR, if `enableAutoApproval` is true.

File: reconcile/templating/renderer.py [hunk 1]
```diff
@@ -136,19 +142,17 @@ def __init__(self, local_path: str, vcs: VCS, mr_manager: MergeRequestManager):
 
     def write(self, outputs: list[TemplateOutput]) -> None:
         self.mr_manager.housekeeping()
+
+        if any([o.input.enable_auto_approval for o in outputs]):
+            auto_approved = [o for o in outputs if o.auto_approved]
+            if auto_approved:
+                self.mr_manager.create_merge_request(auto_approved)
+                return
```
Comment (chassing) [inline reconcile/templating/renderer.py:150]: Something is fishy here. What about all not auto-approved outputs? Is the `return` correct?

Comment (janboll) [reply reconcile/templating/renderer.py:150]: Not auto approved MRs are created after all auto approved ones are created

Comment (janboll) [reply reconcile/templating/renderer.py:150]: (For one collection of course)

Comment (chassing) [reply reconcile/templating/renderer.py:150]: ok

**Review by chassing — APPROVED** [review]

**Comment by janboll** [comment]
Needed to rebase, cause of merge conflict

**Pull request merged**